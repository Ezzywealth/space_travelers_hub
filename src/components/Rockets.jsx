import React from 'react';

const Rockets = () => (
  <div>
    Rockets Rockets Rockets Rockets Rockets Rockets Rockets Rockets Rockets
    Rockets Rockets Rockets Rockets Rockets Rockets
  </div>
);

export default Rockets;
