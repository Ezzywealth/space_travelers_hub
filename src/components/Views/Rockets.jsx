import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../../redux/slices/rockets/rocketSlice';
import Rocket from '../Rocket';

const rocketsUrl = 'https://api.spacexdata.com/v3/rockets';
const Rockets = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets.rockets);
  useEffect(() => {
    if (rockets.length < 1) {
      dispatch(fetchRockets(rocketsUrl));
    }
  }, [dispatch, rockets]);
  return (
    <main className="pb-10">
      <section className="px-4 md:px-20 pt-8 space-y-8 md:space-y-4">
        {rockets.map((rocket) => (
          <Rocket key={rocket.id} rocket={rocket} />
        ))}
      </section>
    </main>
  );
};

export default Rockets;
