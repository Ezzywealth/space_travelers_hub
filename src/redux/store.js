import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './slices/rockets/rocketSlice';
import missionReducer from './slices/missionSlice/missionSlice';
import menuReducer from './slices/menu/menuSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
    menu: menuReducer,
  },
});

export default store;
