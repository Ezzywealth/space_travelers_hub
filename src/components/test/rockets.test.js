import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import TestRenderer, { act } from 'react-test-renderer';
import store from '../../redux/store';
import Rockets from '../Views/Rockets';

describe('Missions', () => {
  it('Snapshot of Rocket Component', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Join button', async () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    const reserveBtn = await screen.findAllByText('Reserve Rocket');
    expect(reserveBtn).toHaveLength(4);
  });

  it('Click on Reserve Rocket button', async () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    act(() => {
      const reserveBtns = screen.getAllByText('Reserve Rocket');
      reserveBtns[0].click();
    });

    const status = await screen.getAllByText('Reserved');
    expect(status).toHaveLength(1);
  });

  it('Click more than one Join button', async () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    act(() => {
      const reserveBtns = screen.getAllByText('Reserve Rocket');
      reserveBtns[2].click();
    });

    act(() => {
      const reserveBtns = screen.getAllByText('Reserve Rocket');
      reserveBtns[1].click();
    });

    const status = await screen.getAllByText('Cancel Reservation');
    expect(status).toHaveLength(3);
  });

  it('No of Reserve Rocket btn after Cancelling a rocket reservation', async () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    act(() => {
      const cancelBtns = screen.getAllByText('Cancel Reservation');
      cancelBtns[0].click();
    });

    const reserveBtns = await screen.getAllByText('Reserve Rocket');
    expect(reserveBtns).toHaveLength(2);
  });

  it('No of Reserved badge after Cancelling 2 rocket reservation', async () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    act(() => {
      const cancelBtns = screen.getAllByText('Cancel Reservation');
      cancelBtns[1].click();
    });

    const reserveBadge = await screen.getAllByText('Reserved');
    expect(reserveBadge).toHaveLength(1);
  });
});
