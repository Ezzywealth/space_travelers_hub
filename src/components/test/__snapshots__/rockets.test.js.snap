// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Snapshot of Rocket Component 1`] = `
<main
  className="pb-10"
>
  <section
    className="px-4 md:px-20 pt-8 space-y-8 md:space-y-4"
  />
</main>
`;
